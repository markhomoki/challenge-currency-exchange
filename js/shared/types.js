export const EUR = 'EUR';
export const GBP = 'GBP';
export const USD = 'USD';

export const CURRENCIES = {
	[EUR]: '€',
	[GBP]: '£',
	[USD]: '$',
};
