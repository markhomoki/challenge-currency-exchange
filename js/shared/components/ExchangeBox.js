import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { CURRENCIES, EUR, GBP, USD } from 'app/types';
import DigitInput from './DigitInput';

export default class ExchangeBox extends React.PureComponent {

	static propTypes = {
		balance: PropTypes.number.isRequired,
		className: PropTypes.string,
		currency: PropTypes.oneOf([EUR, GBP, USD]).isRequired,
	}

	static defaultProps = {
		className: '',
	}

	handleCurrencyChange = (e) => {
		this.props.onCurrencyChange(e.currentTarget.value, this.props.direction);
	}

	renderSymbol = () => {
		return CURRENCIES[this.props.currency] || '';
	}

	render() {
		const {
			balance,
			className,
			currency,
			onCurrencyChange,
			...rest
		} = this.props;

		return (
			<div className={`exchange-item ${className}`}>
				<div className="amount-wrapper">
					<div className="currency-wrapper">
						<select className="currency-select" onChange={this.handleCurrencyChange} value={currency}>
							{_.map(CURRENCIES, (val, cur) => <option key={cur} value={cur}>{cur}</option>)}
						</select>
						<i className="ion-chevron-down" />
					</div>
					<DigitInput {...rest} />
				</div>
				<span className="balance">Balance: {this.renderSymbol()}{balance}</span>
			</div>
		);
	}

}
