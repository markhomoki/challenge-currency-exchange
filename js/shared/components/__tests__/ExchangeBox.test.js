import React from 'react';
import renderer from 'react-test-renderer';
import { USD } from 'app/types';
import ExchangeBox from '../ExchangeBox';

describe('ExchangeBox', () => {
	let wrapper;
	const element = (
		<ExchangeBox
			balance={143.54}
			currency={USD}
			direction="from"
			onInputChange={() => {}}
		/>
	);

	beforeEach(() => {
		wrapper = mount(element);
	});

	it('renders correctly', () => {
		const tree = renderer.create(element).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
