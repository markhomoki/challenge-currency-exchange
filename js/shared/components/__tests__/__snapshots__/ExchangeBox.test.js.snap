// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeBox renders correctly 1`] = `
<div
  className="exchange-item "
>
  <div
    className="amount-wrapper"
  >
    <div
      className="currency-wrapper"
    >
      <select
        className="currency-select"
        onChange={[Function]}
        value="USD"
      >
        <option
          value="EUR"
        >
          EUR
        </option>
        <option
          value="GBP"
        >
          GBP
        </option>
        <option
          value="USD"
        >
          USD
        </option>
      </select>
      <i
        className="ion-chevron-down"
      />
    </div>
    <div
      className="digit-input-wrapper"
    >
      <input
        className="digit-input"
        onChange={[Function]}
        placeholder="0"
        type="text"
        value=""
      />
    </div>
  </div>
  <span
    className="balance"
  >
    Balance: 
    $
    143.54
  </span>
</div>
`;
